src/components/UIShell/SideNav.js

```diff
@@ -50,12 +50,23 @@ export default class SideNav extends React.Component {
 
   state = {
     isExpanded: true,
+    isFocused: false,
   };
 
   handleExpand = () => {
     this.setState(state => ({ isExpanded: !state.isExpanded }));
   };
 
+  handleFocus = () => {
+    this.setState(state => {
+      if (!state.isFocused) return { isFocused: true };
+    });
+  };
+
+  handleBlur = () => {
+    this.setState(() => ({ isFocused: false }));
```

Comment:
Previous state check as `.handleFocus()` does?

Reply:
checking if isFocused is already true and only returning a value if needed. I need to return null for the case where it is.

Reply:
Oh I meant that we'd want to put a similar check here.

Reply:
Doh! Sorry, yeah. will do.